packages/frontend/src/components/Editor/UnsavedChangesAlert.tsx

```diff
@@ -0,0 +1,67 @@
+import {
+  AlertDialog,
+  AlertDialogBody,
+  AlertDialogContent,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogOverlay,
+  Button,
+} from '@chakra-ui/react'
+
+export interface UnsavedChangesAlertProps {
+  cancelRef: React.RefObject<HTMLButtonElement>
+  isOpen: boolean
+  onClose: () => void
+  onLeave?: () => void
+  onStay?: () => void
```

Comment:
actually is onStay being used anywhere? If not, it will just be the same as onClose right?